src/app/config.rs

```diff
@@ -464,6 +464,19 @@ impl Config {
         let initial_connection_window_size =
             parse(strings, ENV_INITIAL_CONNECTION_WINDOW_SIZE, parse_number);
 
+        let tap_disabled = strings
+            .get(ENV_TAP_DISABLED)?
+            .map(|d| !d.is_empty())
+            .unwrap_or(false);
+
+        let control_listener = if tap_disabled {
+            None
+        } else {
+            let addr = parse(strings, ENV_CONTROL_LISTEN_ADDR, parse_socket_addr)?
+                .unwrap_or_else(|| parse_socket_addr(DEFAULT_CONTROL_LISTEN_ADDR).unwrap());
+            Some(Listener { addr })
+        };
```

Comment:
TIOLI: Maybe the logic for parsing the control listener if enabled should be factored out into its own function? `config::parse` is quite long...